### PR TITLE
Add support for `--no-open` flag in `tuist graph`

### PR DESCRIPTION
**Pull request by danrevah**
Resolves https://github.com/tuist/tuist/issues/4636
Request for comments document (if applies):

## Current behaviour

Running

```
$ tuist graph --format=png
```

Generates a PNG and **opens** the image file.

Add a command flag telling `tuist graph` not to open the generated file.

This is useful in cases where you have CI that generates the image (pre-commit) and you don't want the image to show up.

### How to test the changes locally 🧐

> Include a set of steps for the reviewer to test the changes locally.

### Checklist ✅

- [x] The code architecture and patterns are consistent with the rest of the codebase.
- [ ] The changes have been tested following the [documented guidelines](https://docs.tuist.io/contributors/testing-strategy/).
- [ ] The PR includes the label `changelog:added`, `changelog:fixed`, or `changelog:changed`, whenever it should be included in the “Added”, “Fixed” or “Changed” section of the CHANGELOG. Note: when included in the CHANGELOG, the title of the PR will be used as entry, please make sure it is clear and suitable.
- [x] In case the PR introduces changes that affect users, the documentation has been updated.
- [x] In case the PR introduces changes that affect how the cache artifact is generated starting from the `TuistGraph.Target`, the `Constants.cacheVersion` has been updated.

### CODE REVIEW

File: Sources/TuistKit/Services/GraphService.swift [hunk 1]
```diff
@@ -83,13 +84,14 @@ final class GraphService {
         graph: GraphViz.Graph,
         at filePath: AbsolutePath,
         withFormat format: GraphFormat,
-        layoutAlgorithm: LayoutAlgorithm
+        layoutAlgorithm: LayoutAlgorithm,
+        noOpen: Bool = false
```
Comment (danyf90) [inline Sources/TuistKit/Services/GraphService.swift:88]:
```suggestion
        noOpen: Bool
```

File: Sources/TuistKit/Services/GraphService.swift [hunk 2]
```diff
@@ -103,14 +105,18 @@ final class GraphService {
     private func exportPNGRepresentation(
         from graphVizGraph: GraphViz.Graph,
         at filePath: AbsolutePath,
-        layoutAlgorithm: LayoutAlgorithm
+        layoutAlgorithm: LayoutAlgorithm,
+        noOpen: Bool = false
```
Comment (danyf90) [inline Sources/TuistKit/Services/GraphService.swift:109]:
```suggestion
        noOpen: Bool
```

File: Sources/TuistKit/Commands/GraphCommand.swift [hunk 3]
```diff
@@ -72,6 +78,7 @@ struct GraphCommand: AsyncParsableCommand, HasTrackableParameters {
             layoutAlgorithm: layoutAlgorithm,
             skipTestTargets: skipTestTargets,
             skipExternalDependencies: skipExternalDependencies,
+            noOpen: noOpen,
```
Comment (danyf90) [inline Sources/TuistKit/Commands/GraphCommand.swift:81]: I would invert the boolean when passing here so we don't have to write inverted logic
```suggestion
            open: !noOpen,
```

**Review by danyf90 — APPROVED** [review]

**Review by danyf90 — APPROVED** [review]

**Comment by danyf90** [comment]
@allcontributors add @danrevah for code

**Comment by danrevah** [comment]
Fixed failing tests and updated to `open` as suggested

**Review by danyf90 — APPROVED** [review]

**Review by luispadron — APPROVED** [review]

**Pull request merged**